tests/utils/mod.rs

```diff
@@ -34,10 +34,16 @@ pub async fn create_client(
     notify_url: Url,
 ) -> (
     Arc<relay_client::websocket::Client>,
-    UnboundedReceiver<RelayClientEvent>,
+    Receiver<RelayClientEvent>,
 ) {
-    let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
-    let connection_handler = RelayConnectionHandler::new("notify-client", tx);
+    let (tx, mut rx) = tokio::sync::broadcast::channel(8);
```

Comment:
Does the channel capacity size of 8 should be enough?

Reply:
Yeah I think so because this is just how many messages are queued between steps in the tests. Really this should be like no more than 3 or 4 used at a time

Reply:
To clarify, I introduced the broadcast so that messages could be consumed out-of-order. Before you could e.g. consume the response to a method after calling it. But subscribe method now creates 4 messages and different methods in the test cases are able to consume each of these messages and I want to use them out-of-order to make test development easier. Introducing the broadcast allows cloning the stream for these cases to collect messages for later checking.

Reply:
Also I could've modified the relay client to support either mpsc or broadcast, but this seemed more complicated than the simpler forwarder task here. The service doesn't need a broadcast channel anyway so this keeps this just in the tests.

Reply:
Thx for clarifying!